ml-levenberg-marquardt: Fix name of minValues and maxValues in Options interface.

mvernacc opened the pull request:
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- This PR fixes an error in the existing type definitions for [ml-levenberg-marquardt](https://github.com/mljs/levenberg-marquardt)

----

The `LM` function of ml-levenberg-marquardt takes an options argument; `index.d.ts` declares an interface for this options object. The options include arrays of minimum and maximum values for each parameter being fit by the `LM` function. This `index.d.ts` had previously named those arrays `maxValue` and `minValue`. This is an error; the `LM` function uses the `maxValues` and `minValues` fields of the options object. This PR corrects the error.

peterblazejewicz commented:
this is atually a breaking change and this package seems missing proper versioning. The change with options was in v2:
https://github.com/mljs/levenberg-marquardt/compare/v1.0.3...v2.0.0
This package is still at 1.0, the current is 3.11

andrewbranch reviewed (changes requested):
Please update the package version in the comment header per @peterblazejewicz's suggestion. Thanks!

mvernacc commented:
I updated the version in the comment header, and added definitions for v1 and v2 based on the directions [here](https://github.com/DefinitelyTyped/DefinitelyTyped#if-a-library-is-updated-to-a-new-major-version-with-breaking-changes-how-should-i-update-its-type-declaration-package). This is my first time contributing to DefinitelyTyped, so not sure if I did that correctly. Please let me know :)

andrewbranch commented:
Thanks @mvernacc! Adding the older versions is not strictly necessary to have the v3 available, but if they’re there and accurate, that’s great! @peterblazejewicz look ok to you?

peterblazejewicz commented:
we only need to progress, so 3.1 is enough, replacing old one, will review

peterblazejewicz commented on types/ml-levenberg-marquardt/index.d.ts:
```diff
@@ -1,4 +1,4 @@
+// Type definitions for ml-levenberg-marquardt 3.1
 // Project: https://github.com/mljs/levenberg-marquardt#readme
 // Definitions by: m93a <https://github.com/m93a>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
```
```suggestion
// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
```
please drop on next line:
```ts
// TypeScript Version: 2.1
```

mvernacc replied:
Removed in https://github.com/mvernacc/DefinitelyTyped/commit/e0d2fbe8672728c750ce4f556fe6aee34c5b4d22

peterblazejewicz reviewed (approved):
LGTM!
@mvernacc thx!

mvernacc commented:
Ready to merge
